vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -202,7 +202,7 @@ private void initBaseElementsAndListeners() {
         getElement().appendChild(templateElement);
         getElement().appendVirtualChild(container);
 
-        getElement().addEventListener("opened-changed", event -> {
+        addOpenedChangeListener(event -> {
```

Comment:
An "opened-changed" event only works if the Flow component doesn't get detached from its parent while the `<vaadin-notification>` Web Component is showing. Then we could rely on the DOM event to take care of getting this callback invoked.

But if the Flow component gets detached before the notification actually closes, the Web Component becomes an "orphan" without an active Flow counterpart and the following "opened-changed" event never reaches the callback.